git-push-fork-to-upstream-branch

```diff
@@ -15,10 +15,10 @@ fi
 
 SOURCE_GH_USER=$(echo "$BRANCH_SPEC" | awk -F: '{print $1}')
 SOURCE_BRANCH=$(echo "$BRANCH_SPEC" | awk -F: '{print $2}')
-REPO_NAME=$(basename $(git rev-parse --show-toplevel))
+REPO_NAME=$(git remote get-url --push origin | cut -d '/' -f 2 | cut -d '.' -f 1)
 
 git remote remove fork-to-test || echo "Added new remote fork-to-test"
-git remote add fork-to-test https://github.com/$SOURCE_GH_USER/$REPO_NAME.git
+git remote add fork-to-test git@github.com:$SOURCE_GH_USER/$REPO_NAME.git
```

Comment:
What effect does this change have?

Reply:
I tried this out locally, and it asked for me to enter my passphrase for my ssh key. I must have state saved locally for authenticating to GitHub via https, but not for ssh; I always clone using https. And I assume that you have local state set up for making ssh connections to GitHub. We might need to make this an option.

Reply:
Bingo.